cmd/publicize/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"k8s.io/test-infra/prow/githubeventserver"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+
+	"k8s.io/test-infra/pkg/flagutil"
+	"k8s.io/test-infra/prow/config/secret"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+)
+
+type Config struct {
+	Bonds []Bond `json:"bonds,omitempty"`
+}
+
+func (c *Config) validate() error {
+	var errs []error
+	for index, bond := range c.Bonds {
+		if err := bond.validate(); err != nil {
+			errs = append(errs, fmt.Errorf("bond[%d]: %w", index, err))
+		}
+	}
+
+	return kerrors.NewAggregate(errs)
+}
+
+// bond holds the information of which repos are bonded to each other.
+type Bond struct {
+	From string `json:"from"`
+	To   string `json:"to"`
+}
+
+func (b *Bond) validate() error {
+	for _, value := range []string{b.From, b.To} {
+		if len(strings.Split(value, "/")) != 2 {
+			return fmt.Errorf("%s should be in org/repo format", value)
+		}
+	}
+	return nil
+}
+
+type options struct {
+	configPath               string
+	gitName                  string
+	gitEmail                 string
+	bonds                    map[string]string
+	dryRun                   bool
+	githubEventServerOptions githubeventserver.Options
+	github                   prowflagutil.GitHubOptions
+	git                      prowflagutil.GitOptions
+
+	webhookSecretFile string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+
+	fs.StringVar(&o.configPath, "config-path", "/etc/config/config.yaml", "Path to publicize configuration.")
```

Comment:
I'm a little confused about the usage here. This wouldn't read normal Prow `plugins.yaml`?

Reply:
This will ready the publicize configuration file. Not related to prow.

Reply:
Well the description says it's an external Prow plugin. Other external prow plugins seem to have config in that file... But yeah, I guess I understand, this is a *real* external plugin, it does not make sense for the `test-infra` config struct to carry this config.